src/structured-data/handler.js

```diff
@@ -125,14 +125,21 @@ export async function processStructuredData(baseURL, context, pages) {
 }
 
 export async function opportunityAndSuggestions(auditUrl, auditData, context) {
+  const { log } = context;
+
+  // Check if audit was successful
+  if (auditData.auditResult.success === false) {
```

Comment:
this could possibly be applicable to all post processors. should this be handled in the common code?

Reply:
I agree. This pattern is currently only used by `apex`, `backlinks`, `canonical`, `sitemap` and `structured-data` audits, but can be adopted by all of them.

I'll create a follow up JIRA ticket to update the implementation of the post processor logic and to also update the README to give guidance on how this pattern should be used.